### PR TITLE
OPENEUROPA-2391: Add EU footer to corporate blocks.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

The scope of this issue is to add the EU footer data to the corporate footer. This should go in version 2.x of the oe_corporate_blocks.

The EU specific footer data can be retrieved from https://europa.eu/european-union/index_en , all language versions are available by using the language switcher.
This data fits directly into the example layout defined in ECL here: https://ec.europa.eu/component-library/playground/eu/?path=/story/page-structure-footer--corporate
### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

**Review by nagyad — CHANGES_REQUESTED** [review]
We need to retrieve the complete sentences for the first 4 links.

File: modules/oe_corporate_blocks_demo/config/install/block.block.eufooterblock.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+uuid: ad7c9aef-ee7a-48d0-a38b-02ca5f5e0658
```
Comment (nagyad) [inline modules/oe_corporate_blocks_demo/config/install/block.block.eufooterblock.yml]: We can't have this.

File: src/Plugin/Block/FooterBlockBase.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory')
```
Comment (nagyad) [inline src/Plugin/Block/FooterBlockBase.php]: Could we have this in the order of the arguments?

File: templates/oe-corporate-blocks-eu-footer.html.twig [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * @ingroup themeable
+ */
+#}
+
+<div class="footer__corporate-top--eu">
+  {% if corporate_footer.contact %}
+    <ul class="footer__list">
+      {% for menu_item in corporate_footer.contact %}
+        <li class="footer__list-item">
+          {{ menu_item|raw }}
```
Comment (nagyad) [inline templates/oe-corporate-blocks-eu-footer.html.twig]: Why we need the |raw here?

File: tests/features/corporate-blocks.feature [hunk 4]
```diff
@@ -51,3 +51,74 @@ Feature: Corporate blocks feature
       | path  |
       | home  |
       | login |
+
+  Scenario Outline: On any pages we have correct urls in EU corporate footer block for many languages
```
Comment (nagyad) [inline tests/features/corporate-blocks.feature]: Urls are correctly shown in the EU corporate footer block when page and language are changed.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: templates/oe-corporate-blocks-eu-footer.html.twig (hunk 3 above)
Comment (sergepavle) [reply templates/oe-corporate-blocks-eu-footer.html.twig]: because `menu_item` contains a pure HTML (not a renderable array).

Comment (nagyad) [reply templates/oe-corporate-blocks-eu-footer.html.twig]: This variable will not be escaped.

**Review by nagyad — DISMISSED** [review]

**Review by upchuk — DISMISSED** [review]
Upgrade path to import the new config? Here or EWCMS?

File: templates/oe-corporate-blocks-eu-footer.html.twig (hunk 3 above)
Comment (sergepavle) [reply templates/oe-corporate-blocks-eu-footer.html.twig]: corrected

File: translations/oe_corporate_blocks-sl.po [hunk 5]
```diff
@@ -1,16 +1,132 @@
+msgid ""
+msgstr ""
 
-msgid "European Commission"
-msgstr "Evropska komisija"
```
Comment (ademarco) [inline translations/oe_corporate_blocks-sl.po:3]: Did this string just got removed from the translations?

File: src/Plugin/Block/EcFooterBlock.php [hunk 6]
```diff
@@ -5,58 +5,19 @@
...
 use Drupal\Core\Cache\CacheableMetadata;
-use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Provides the corporate footer block.
  *
  * @Block(
  *   id = "oe_footer",
```
Comment (ademarco) [inline src/Plugin/Block/EcFooterBlock.php]: Let's use this occasion to properly prefix plugin names with the module's name, let's change this one into

```
oe_corporate_blocks_ec_footer
```

File: src/Plugin/Block/EuFooterBlock.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+
+/**
+ * Provides the corporate EU footer block.
+ *
+ * @Block(
+ *   id = "oe_eu_footer",
```
Comment (ademarco) [inline src/Plugin/Block/EuFooterBlock.php]: Same here, please change into:
```
oe_corporate_blocks_eu_footer
```

Also change this in the demo module configuration.

**Review by ademarco — CHANGES_REQUESTED** [review]
@sergepavle  a few change requests.

File: translations/oe_corporate_blocks-sl.po (hunk 5 above)
Comment (sergepavle) [reply translations/oe_corporate_blocks-sl.po:3]: Yes. it is fixed also for ET language.

File: oe_corporate_blocks.post_update.php [hunk 8]
```diff
@@ -40,3 +43,29 @@ function oe_corporate_blocks_post_update_10001(&$sandbox): void {
   $string = $storage->findString(['source' => 'European Commission website']);
   $storage->delete($string);
 }
+
+/**
+ * Update existing blocks for EC corporate footer and import new configs.
+ */
+function oe_corporate_blocks_post_update_20002(&$sandbox): void {
+  // Replacing plugin for existing footer.
```
Comment (ademarco) [inline oe_corporate_blocks.post_update.php]: We should clear block plugin cache or updb will fail with:
```
>  [notice] Update started: oe_corporate_blocks_post_update_20002
>  [error]  Plugin (oe_footer) instance class "Drupal\oe_corporate_blocks\Plugin\Block\FooterBlock" does not exist. 
>  [error]  Update failed: oe_corporate_blocks_post_update_20002 
```

Add this:

```php
function oe_corporate_blocks_post_update_20002(&$sandbox): void {
  // Clear cached block definition as we have renamed EC footer base class.
  \Drupal::service('plugin.manager.block')->clearCachedDefinitions();

  // Replacing plugin for existing footer.
...
```

**Review by ademarco — CHANGES_REQUESTED** [review]

**Review by nagyad — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**